src/formatDate.js

```diff
@@ -8,7 +8,86 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let result = '';
+  const oldDate = date;
+  const newFormat = toFormat;
+  let array = [];
+  let year = '';
+  let yearBig = '';
+  const newArray = [];
+  let indexYear = 0;
+  let indexYearbig = 0;
+
+  const [, , , oldSeparator] = fromFormat;
+  const [, , , newSeparator] = newFormat;
+
+  array = oldDate.split(oldSeparator);
+
+  if (newFormat.includes('YYYY') && fromFormat.includes('YYYY')) {
+    for (const n of newFormat) {
+      const searchInd = fromFormat.indexOf(n);
+
+      newArray.push(array[searchInd]);
```

Comment:
The `newArray.pop()` operation removes the last element, which might not be necessary or correct depending on the format. Ensure that this operation aligns with the intended format transformation.